Dockerfile-linux.template

```diff
@@ -107,14 +127,57 @@ RUN set -eux; \
 	\
 # save the timestamp from the tarball so we can restore it for reproducibility, if necessary (see below)
 	SOURCE_DATE_EPOCH="$(stat -c '%Y' /usr/local/go)"; \
+{{ ) else ( -}}
+# before we get too far, let's validate that our "bootstrap" Go works
+	export GOROOT_BOOTSTRAP=/usr/local/goroot-bootstrap; \
+# TODO figure out why QEMU's user-mode emulation with Go 1.23.0 causes "can't start telemetry child process: fork/exec /usr/local/go/bin/go: invalid argument" (it works if we run it a second time, but then later "go build" fails to fork/exec the compile process 🙃)
```

Comment:
because of this "fun" with Go 1.23 + QEMU, I haven't done much testing of this on architectures other than amd64 and i386, so that's a notable blind spot that's probably worth more testing before we consider merging